tool-openssl/enc.cc

```diff
@@ -0,0 +1,202 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/evp.h>
+#include <stdio.h>
+#include <string.h>
+#include <algorithm>
+#include <iostream>
+#include "../tool/internal.h"
+#include "internal.h"
+
+#define BUF_SIZE 1024
+
+static const argument_t kArguments[] = {
+    // General options
+    {"-help", kBooleanArgument, "Display option summary"},
+    {"-in", kOptionalArgument, "Input file, default stdin"},
+    {"-out", kOptionalArgument, "Output file, default stdout"},
+    {"-e", kBooleanArgument, "Encrypt"},
+    {"-d", kBooleanArgument, "Decrypt"},
+    {"-K", kOptionalArgument, "Raw key to use, in hex form"},
+    {"-iv", kOptionalArgument, "IV to use, in hex form"},
+    {"-aes-128-cbc", kExclusiveBooleanArgument, "Supported cipher"},
+    {"", kOptionalArgument, ""}};
+
+static bool HexToBinary(uint8_t *buffer, const std::string &hex_string,
+                        unsigned int size) {
+  // First validate that the string contains only valid hex characters
+  for (char c : hex_string) {
+    if (!OPENSSL_isxdigit(c)) {
+      return false;
+    }
+  }
+
+  if (hex_string.size() != size * 2) {
+    return false;
+  }
+
+  BIGNUM *raw = NULL;
+  if (BN_hex2bn(&raw, hex_string.c_str()) == 0) {
+    return false;
+  }
+
+  int ret = BN_bn2binpad(raw, buffer, size);
```

Comment:
Can we use `BN_bn2bin_padded` instead? Seems like that's easier to use (judging from the function documentation).

https://github.com/aws/aws-lc/blob/ec39cb332cd0c01eab5ed8d17d282811f8f52a54/include/openssl/bn.h#L1015C9-L1016